test/box_test.dart

```diff
@@ -168,6 +170,16 @@ void main() {
     expect(removed, equals(3));
   });
 
+  test("write in tnx works", () {
+    int count;
+    write_func() {
+      box.putMany(simple_items);
+    }
+    store.runInTransaction(TxMode.Write, write_func);
+    count = box.count();
+    expect(count, equals(6));
+  });
+
```

Comment:
Probably found a bug while testing:  
`test("failing in recursive tnx", () {
    store.runInTransaction(TxMode.Write, () {
      //should throw code10001
      List<int> ids = store.runInTransaction(TxMode.Read, () {
        return box.putMany(simple_items);
      });
      expect(ids.length, equals(6));
    });
  });`
this test passes but shouldn't it fail and throw an exception?

Reply:
Or is that valid because only the outermost transaction "counts" on DB level -> thus beeing a write transaction?

Reply:
> Or is that valid because only the outermost transaction "counts" on DB level -> thus beeing a write transaction?

That's most probably what happens but I'm not sure whether that should be considered "valid". 
In any case it's a programming error to write in a read transaction, so is not really such a big deal but I'll report it in the core. Ideally, it would be refused even if the top-level is a write Tx...

Reply:
BTW, if you don't mind could you use `tx` or `txn` instead of `tnx`. While valid, the rest of the code/c-api uses txn

Reply:
> Good job noticing this :)

Thanks 😄  

>BTW, if you don't mind could you use tx or txn instead of tnx. While valid, the rest of the code/c-api uses txn

Yeah I think that one was only a typo 👍